File: leetcode/golang/102_binary_tree_level_order_traversal.go

type TreeNode struct {
    Val int
    Left *TreeNode
    Right *TreeNode
}

 func levelOrder(root *TreeNode) [][]int {
    var queue [] *TreeNode
    var res [][]int
    if root!= nil {
        queue = append(queue, root)
    }
    for len(queue) != 0 {
        var tmp []int
        for i := len(queue); i > 0; i-- {
            tmpNode := queue[0]
            queue = queue[1:]
            tmp = append(tmp, tmpNode.Val)
            if tmpNode.Left != nil {
                queue = append(queue, tmpNode.Left)
            }
            if tmpNode.Right != nil {
                queue = append(queue, tmpNode.Right)
            }
        }
        res = append(res, tmp)
    }
    return res
}

File: leetcode/c++/102_binary_tree_level_order_traversal.cpp
// https://leetcode.cn/problems/binary-tree-level-order-traversal/description

#include <bits/stdc++.h>
using namespace std;


struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        queue<TreeNode *> tree_queue;
        vector<vector<int>> res;
        if (root != nullptr) tree_queue.push(root);
        while (!tree_queue.empty()) {
            vector<int> tmp;
            for (int i = tree_queue.size(); i > 0; i--) {
                TreeNode* tmp_node = tree_queue.front();
                tree_queue.pop();
                tmp.push_back(tmp_node->val);
                if (tmp_node->left != nullptr) tree_queue.push(tmp_node->left);
                if (tmp_node->right != nullptr) tree_queue.push(tmp_node->right);  
            }
            res.push_back(tmp);
        }
        return res;
    }
};